dsp/dsp.py

```diff
@@ -0,0 +1,254 @@
+import numpy as np
+import scipy.signal as sig
+
+window_types = [
+    'boxcar',
+    'bartlett',
+    'hann',
+    'blackman',
+    'blackmanharris'
+]
+window_sizes = [64, 128, 256, 512, 1024, 2048, 4096]
+
+def compute_cola_window(config):
+    """
+    Using 'window_type' and 'window_size' values from 'config', computes 
+    desired window and corresponding hop size to achieve constant-overlap-add 
+    (COLA) property. Note that with some window - window size combinations the
+    quality of the COLA reconstruction is not perfect.
+    """
+    window_type = config['window_type']
+    window_size = config['window_size']
+    assert(window_size >= 64 and (window_size & (window_size - 1)) == 0)
```

Comment:
Not sure why `(window_size & (window_size - 1)) == 0` is needed.

Reply:
we want `window_size` to be a power of two. this is a little cryptic but that's all it's doing. i can put a comment about it in the docstring if you'd like